pkg/gardenlet/controller/shoot/shoot_control_reconcile.go

```diff
@@ -626,10 +646,34 @@ func (r *shootReconciler) runReconcileShootFlow(ctx context.Context, o *operatio
 			Dependencies: flow.NewTaskIDs(waitUntilWorkerReady, deployManagedResourcesForAddons, deployManagedResourceForCloudConfigExecutor),
 		})
 
+		deployExtensionResourcesAfterKAPI = g.Add(flow.Task{
+			Name:         deployExtensionAfterKAPIMsg,
+			Fn:           flow.TaskFn(botanist.DeployExtensionsAfterKubeAPIServer).RetryUntilTimeout(defaultInterval, defaultTimeout),
+			Dependencies: flow.NewTaskIDs(deployReferencedResources, initializeShootClients),
+		})
+		waitUntilExtensionResourcesAfterKAPIReady = g.Add(flow.Task{
+			Name:         waitExtensionAfterKAPIMsg,
+			Fn:           botanist.Shoot.Components.Extensions.Extension.WaitAfterKubeAPIServer,
+			Dependencies: flow.NewTaskIDs(deployExtensionResourcesAfterKAPI),
+		})
+
 		hibernateControlPlane = g.Add(flow.Task{
 			Name:         "Hibernating control plane",
 			Fn:           flow.TaskFn(botanist.HibernateControlPlane).RetryUntilTimeout(defaultInterval, 2*time.Minute).DoIf(o.Shoot.HibernationEnabled),
-			Dependencies: flow.NewTaskIDs(initializeShootClients, deploySeedMonitoring, deploySeedLogging, deployClusterAutoscaler),
+			Dependencies: flow.NewTaskIDs(initializeShootClients, deploySeedMonitoring, deploySeedLogging, deployClusterAutoscaler, waitUntilExtensionResourcesAfterKAPIReady),
+		})
+
+		// logic is inverted here
+		// extensions that are deployed before the kube-apiserver are hibernated after it
+		hibernateExtensionResourcesAfterKAPIHibernation = g.Add(flow.Task{
```

Comment:
IIUC the proposal, shouldn't similar steps be used for the `DeployExtensionsAfterKubeAPIServer` tasks if the `o.Shoot.HibernationEnabled` is true, before scaling down the kube-apiserver?

Reply:
Yes, this is done [here](https://github.com/gardener/gardener/pull/6999/files#diff-b08cfdfda24a4eb57a4f1f324e3747cb5b65932440624fbf7e2403b76e8c26b5R649-R658) with a single step because of the flow

deploy kapi -> deploy extensions -> do not hibernate kapi
deploy kapi -> deploy(hibernate) extensions -> hibernate kapi

This is also why we determine the message [here](https://github.com/gardener/gardener/pull/6999/files#diff-b08cfdfda24a4eb57a4f1f324e3747cb5b65932440624fbf7e2403b76e8c26b5R116).